DatadogLogs/Tests/WebViewLogReceiverTests.swift

```diff
@@ -0,0 +1,203 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+import DatadogInternal
+
+@testable import DatadogLogs
+
+class WebViewLogReceiverTests: XCTestCase {
```

Comment:
I'm sorry but I don't see the violation? It uses the `PassthroughCoreMock` because we are in the module unit tests, and the `WebViewLogReceiver` is set as a receiver of the core. The `WebViewLogReceiver` has no other API that `receive`. I could do the following:
```swift
messageReceiver.receive(
    message: .baggage(...),
    from: core
)
```
But that is essentially the same, no?

Reply:
It is totally fine to use `PassthroughCoreMock` 👍 as we don't care about the real core behaviour in these tests. For the same reason, yes we should test:
```swift
messageReceiver.receive(message:from:)
```
and not `.send(message:else:)` on core mock. 

Otherwise we cannot test all behaviours without re-implementing real logic into `PassthroughCoreMock`. Take an example: _"given message addressed to other feature, web view log receiver rejects it"_ - it is a matter of asserting `false` returned from `receive(message:from:)` vs basing assertion on [re-implemented fallback logic](https://github.com/DataDog/dd-sdk-ios/blob/336d1dcf78402e7a00310a5b081007aa7210a344/TestUtilities/Mocks/CoreMocks/PassthroughCoreMock.swift#L108) in mock.

Exactly this change of paradigm was the actual thing that surfaced the `FeatureBaggage` propagation problems in https://github.com/DataDog/dd-sdk-ios/pull/1502/files#r1341160540. For that reason, I'm advocating for less dependency on mocks, more edge case coverage and more fuzzy testing.

Reply:
Ok, yes it makes sense! Thanks for the clarification 🙏